From 84b955fbd61b30b8d1b5804b2a112209e3bcaf2e Mon Sep 17 00:00:00 2001
From: Robert Kirkman <robert.kirkman@ttu.edu>
Date: Wed, 2 Mar 2022 07:55:11 -0600
Subject: [PATCH 1/1] add BBB to Makefile

---
 Makefile | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/Makefile b/Makefile
index 3c7898a..983710a 100644
--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,9 @@ NON_MATCHING ?= 1
 # Build and optimize for Raspberry Pi(s)
 TARGET_RPI ?= 0
 
+# Build and optimize for BeagleBone Black
+TARGET_BBB ?= 0
+
 # Build for Emscripten/WebGL
 TARGET_WEB ?= 0
 
@@ -219,6 +222,10 @@ ifeq ($(TARGET_RPI),1) # Define RPi to change SDL2 title & GLES2 hints
       VERSION_CFLAGS += -DUSE_GLES
 endif
 
+ifeq ($(TARGET_BBB),1) # Define BBB to change SDL2 title & GLES2 hints
+      VERSION_CFLAGS += -DUSE_GLES
+endif
+
 ifeq ($(OSX_BUILD),1) # Modify GFX & SDL2 for OSX GL
      VERSION_CFLAGS += -DOSX_BUILD
 endif
@@ -376,6 +383,11 @@ ifeq ($(TARGET_RPI),1)
         endif
 endif
 
+# BeagleBone Black. Its architecture is not identical to any RPi
+ifeq ($(TARGET_BBB),1)
+        OPT_FLAGS := -march=armv7-a -marm -mfpu=neon -mtune=cortex-a8 -O3
+endif
+
 # File dependencies and variables for specific files
 include Makefile.split
 
@@ -528,6 +540,8 @@ else ifeq ($(findstring SDL,$(WINDOW_API)),SDL)
     BACKEND_LDFLAGS += -lglew32 -lglu32 -lopengl32
   else ifeq ($(TARGET_RPI),1)
     BACKEND_LDFLAGS += -lGLESv2
+  else ifeq ($(TARGET_BBB),1)
+    BACKEND_LDFLAGS += -lGLESv2
   else ifeq ($(OSX_BUILD),1)
     BACKEND_LDFLAGS += -framework OpenGL $(shell pkg-config --libs glew)
   else
@@ -668,6 +682,9 @@ else ifeq ($(WINDOWS_BUILD),1)
 else ifeq ($(TARGET_RPI),1)
   LDFLAGS := $(OPT_FLAGS) -lm $(BACKEND_LDFLAGS) -no-pie
 
+else ifeq ($(TARGET_BBB),1)
+  LDFLAGS := $(OPT_FLAGS) -lm $(BACKEND_LDFLAGS) -no-pie
+
 else ifeq ($(OSX_BUILD),1)
   LDFLAGS := -lm $(PLATFORM_LDFLAGS) $(BACKEND_LDFLAGS) -lpthread
 
-- 
2.30.2
